sdk/face/ai-vision-face-rest/src/models.ts

```diff
@@ -0,0 +1,144 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+/** Request for creating liveness session. */
+export interface CreateLivenessSessionContent {
+  /** Type of liveness mode the client should follow. */
+  livenessOperationMode: LivenessOperationMode;
+  /** Whether or not to allow a '200 - Success' response body to be sent to the client, which may be undesirable for security reasons. Default is false, clients will receive a '204 - NoContent' empty body response. Regardless of selection, calling Session GetResult will always contain a response body enabling business logic to be implemented. */
+  sendResultsToClient?: boolean;
+  /** Whether or not to allow client to set their own 'deviceCorrelationId' via the Vision SDK. Default is false, and 'deviceCorrelationId' must be set in this request body. */
+  deviceCorrelationIdSetInClient?: boolean;
+  /** Unique Guid per each end-user device. This is to provide rate limiting and anti-hammering. If 'deviceCorrelationIdSetInClient' is true in this request, this 'deviceCorrelationId' must be null. */
+  deviceCorrelationId?: string;
+  /** Seconds the session should last for. Range is 60 to 86400 seconds. Default value is 600. */
+  authTokenTimeToLiveInSeconds?: number;
+}
+
+export interface ICreateLivenessWithVerifySessionContentParametersPartDescriptor {
+  name: "Parameters";
+  body: CreateLivenessSessionContentForMultipart;
+}
+
+export class CreateLivenessWithVerifySessionContentParametersPartDescriptor {
+  get name(): string {
+    return "Parameters";
+  }
+
+  body: CreateLivenessSessionContentForMultipart;
+
+  constructor(descriptor: ICreateLivenessWithVerifySessionContentParametersPartDescriptor) {
+    this.body = descriptor.body;
+  }
+}
```

Comment:
why we need this wrapper over the actual part definition? the RLC mfd implementation doesn't mention this https://github.com/Azure/autorest.typescript/blob/main/packages/typespec-ts/test/integration/generated/payload/multipart/src/models.ts#L145 ? 

@timovv @joheredi would you mind explain a little? have we changed the design, if so, I am curious why?

Reply:
no, this wrapper doesn't get generated. Looks to be customization?

Reply:
Yes it's customized. This mainly for `CreateLivenessWithVerifySessionContentVerifyImagePartDescriptor`. There is an optional field `filename` which is actually required by backend but makes no sense to clients, so we tend to keep it optional and give it a default value. However, it's impossible for TypeScript interfaces to set default values, so my solution is making it a class.

Reply:
I have a question, if customer is calling this API directly, will they need to pass the filename? if yes, so should RLC and maybe other clients, 

Another point is, we are not suppose to support default value in client side according to the SDK guideline https://azure.github.io/azure-sdk/general_implementation.html#parameter-validation

Reply:
The `filename` field is a part of the [MFD part descriptor](https://github.com/Azure/azure-sdk-for-js/blob/5071f20655aa16cf012c93c17a150af792f484fe/sdk/core/core-client-rest/src/multipart.ts#L44). After reading more documentations, I think the correct solution is updating the `getContentDisposition` function in [multipart.ts](https://github.com/Azure/azure-sdk-for-js/blob/5071f20655aa16cf012c93c17a150af792f484fe/sdk/core/core-client-rest/src/multipart.ts#L143-L151) to assign default filenames if not specified to align the behavior of [JS standard library](https://developer.mozilla.org/en-US/docs/Web/API/FormData/append#filename). I assume you just have to add one more condition:

```js
  let filename: string | undefined = undefined;
  if (descriptor.filename) {
    filename = descriptor.filename;
  } else if (typeof File !== "undefined" && descriptor.body instanceof File) {
    const filenameFromFile = (descriptor.body as File).name;
    if (filenameFromFile !== "") {
      filename = filenameFromFile;
    }
  } else if (isBinaryBody(descriptor.body)) {
    filename = "blob";
  }
```

Could you please help with this? So I can revert this customization.

Reply:
Hmm, it's interesting, for one hand, we should not set the default value per SDK general guideline, for the other hand, we should be consistent if customer is switching between FormData and our own implementation. 

Also, I am not sure if we really support Blob or not in core, it says filename default to blob only applies to Blob but the above code didn't check if it's Blob instance, is it possible there's could be a type that is either FIle or Blob type? what's the default name then?

And there's also a note in the FormData description, 
![image](https://github.com/Azure/azure-sdk-for-js/assets/55688292/ab298fbf-5f3a-487a-936d-d76845a0b651)
though I am not sure why, but if all browsers adhere to that guideline, why would it be vary from browser to browser ? 

if they don't, why would we?

@joheredi @timovv @xirzec @bterlson any thoughts here?

Reply:
With the patch above applied, these statements work well:

- `body: readFileSync('detection1.jpg') // Buffer, inheriting Uint8Array`
- `body: new Blob([readFileSync('detection1.jpg')]) // Blob`
- `body: new File([readFileSync('detection1.jpg')], 'detection1.jpg') // File, inheriting Blob`

Because supported octet-stream types ([string | Uint8Array | ReadableStream<Uint8Array> | NodeJS.ReadableStream | File](https://github.com/chungshengfu/azure-sdk-for-js/blob/73741095d84a8173dbfc60a5b0a8bf912859db1c/sdk/face/ai-vision-face-rest/src/models.ts#L35-L45)) are easily convertible to Blob, the patch above checks `isBinaryBody` instead of `instanceof Blob`.

I think the reason why it says the default filename for Blob can vary among browsers is because it "doesn't matter". It just needs some value in the content-disposition header.

Please note that it's very possible that in vision applications, users send images captured by cameras, so it's definitely image data without a filename.

Reply:
I am curious what would the backend receive if in the case of images captured by cameras?

Reply:
@timovv  I like the customization idea with policy here, just curious,, do you think if the policy should be per client or per operations?